pkg/util/arm/arm.go

```diff
@@ -11,26 +11,45 @@ import (
 	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/sirupsen/logrus"
 
 	"github.com/Azure/ARO-RP/pkg/api"
-	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
 	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
 )
 
-func (m *manager) deployARMTemplate(ctx context.Context, rg string, tName string, t *arm.Template, params map[string]interface{}) error {
-	m.log.Printf("deploying %s template", tName)
+type Manager interface {
+	Deploy(ctx context.Context, resourceGroup string, deploymentName string, template *Template, parameters map[string]interface{}) error
+}
+
+type manager struct {
+	log         *logrus.Entry
+	deployments features.DeploymentsClient
+}
+
+func NewManager(azEnv *azure.Environment, log *logrus.Entry, subscriptionID string, authorizer autorest.Authorizer) Manager {
+	// TODO(mj): we are not able to pass env.Core here as it causes circular imports
+	// with resource generators. This needs to be fixed by re-organizing the code.
+	return &manager{
+		log:         log,
+		deployments: features.NewDeploymentsClient(azEnv, subscriptionID, authorizer),
+	}
+}
+
+func (m *manager) Deploy(ctx context.Context, resourceGroupName string, deploymentName string, template *Template, parameters map[string]interface{}) error {
+	m.log.Printf("deploying %s template", deploymentName)
 
-	err := m.deployments.CreateOrUpdateAndWait(ctx, rg, deploymentName, mgmtfeatures.Deployment{
+	err := m.deployments.CreateOrUpdateAndWait(ctx, resourceGroupName, deploymentName, mgmtfeatures.Deployment{
 		Properties: &mgmtfeatures.DeploymentProperties{
-			Template:   t,
-			Parameters: params,
+			Template:   template,
+			Parameters: parameters,
 			Mode:       mgmtfeatures.Incremental,
 		},
 	})
 
 	if azureerrors.IsDeploymentActiveError(err) {
-		m.log.Printf("waiting for %s template to be deployed", tName)
-		err = m.deployments.Wait(ctx, rg, deploymentName)
+		m.log.Printf("waiting for %s template to be deployed", deploymentName)
+		err = m.deployments.Wait(ctx, resourceGroupName, deploymentName)
 	}
```

Comment:
the more templates we start deploying with the same name, the wronger this code gets.  This should cancel the active deployment and retry.